### DISCUSSION TITLE
How do I display the track name for a specific track.

### DISCUSSION THREAD

**Discussion by jam-sys** [tttapa/Control-Surface#770] [Q&A]
I have build a controlsurface with one fader, one display and a number of buttons, like the Behringer X Touch One.
Most of the functions work, but I have troubles to display the track name for a selected track.

I tried following array, but this prints all file names on top of each other:
```
MCU::LCDDisplay lcddisps[] {
  // track (1),  line 1,  position (0, 20), font size (1)
  {display, lcd, 1, 1, {0, 20}, 1, WHITE},
  {display, lcd, 2, 1, {0, 20}, 1, WHITE},
  {display, lcd, 3, 1, {0, 20}, 1, WHITE},
  {display, lcd, 4, 1, {0, 20}, 1, WHITE},
  {display, lcd, 5, 1, {0, 20}, 1, WHITE},
  {display, lcd, 6, 1, {0, 20}, 1, WHITE},
  {display, lcd, 7, 1, {0, 20}, 1, WHITE},
  {display, lcd, 8, 1, {0, 20}, 1, WHITE},
};
```
What is the correct code to do this?

**Comment by Veroledez** [comment]
Hola, mira este post
https://github.com/tttapa/Control-Surface/issues/281#issuecomment-699483751

**Comment by tttapa** [comment]
> I tried following array, but this prints all file names on top of each other:

That's because they all have the same location: make sure each element in the array has a different x coordinate, e.g. give the first one a position of `{0, 20}`, the second `{40, 20}`, etc.

**Reply by jam-sys** [reply]
Thanks Peter, for your reply.
The point is that I need to display the information  for a single track. I have one display for multiple tracks. I use buttons to step trough the tracks,
similar to the Behringer X Touch One.

**Reply by tttapa** [reply]
In that case you can use the constructor that accepts a Bank: https://tttapa.github.io/Control-Surface-doc/Doxygen/d7/dae/classMCU_1_1LCDDisplay.html#a845baab617ad6f3f97055f75cae26dfc  
If the second bank is selected, the display will show the name of the second track, etc.

```cpp
MCU::LCDDisplay lcddisp {
  // track (1),  line (1),  position (0, 20), font size (1)
  display, lcd, bank, 1, 1, {0, 20}, 1, WHITE,
};
```

**Reply by jam-sys** [reply]
OK, Its makes sense. 
Thanks for your prompt reply.
And thanks a lot for your excellent library.
Are you located in the Netherlands?

**Reply by tttapa** [reply]
Close, I'm from Belgium.

**Reply by jam-sys** [reply]
It works great for the track names, solo, mute, record etc.
The last function I need, is a routine similar to BitmapDisplay, to control leds in the buttons.
Can you give me an idea how to implement that?
I tried to modify BitmapDisplay.hpp but did not succeed.

**Comment by pukulsesuatu** [comment]
![IMG_20220927_192556](https://user-images.githubusercontent.com/532309/193393594-05b323ce-cea0-4b0c-876d-c4bf5067450e.jpg) I am following the instruction but same result, no trackname display at oled , I dont know why
